s4e-backend/src/main/java/pl/cyfronet/s4e/bean/Group.java

```diff
@@ -0,0 +1,44 @@
+package pl.cyfronet.s4e.bean;
+
+import lombok.Builder;
+import lombok.Data;
+import org.hibernate.annotations.NaturalId;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotEmpty;
+import java.util.List;
+
+@Entity
+@Data
+@Builder
+@Table(uniqueConstraints = @UniqueConstraint(columnNames = {"name", "slug", "institution_id"}))
+public class Group {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotEmpty
+    private String name;
+
+    @NotEmpty
+    @NaturalId
```

Comment:
Don't you have to set mutable=true for @NaturalId? It will change on name change.